### PR TITLE
Warning cleanup

### PR DESCRIPTION
**Pull request by wesbland**
Another round of warnings cleanups all over the place. This started with things reported by ICC, but expanded beyond that.

### CODE REVIEW

**Comment by wesbland** [comment]
test:jenkins/ch4/most
test:jenkins/ch3/most

File: src/mpid/ch4/include/mpidpre.h [hunk 1]
```diff
@@ -244,6 +244,7 @@ typedef enum {
 } MPIDIG_win_info_accumulate_ops;
 
 typedef enum {
+    MPIDIG_ACCU_OP_FIRST = 0,
```
Comment (pavanbalaji) [inline src/mpid/ch4/include/mpidpre.h]: The naming convention for this enum is super messed up.  For now, can you change this to `MPIDIG_ACCU_OP_SHIFT_FIRST` to match `MPIDIG_ACCU_OP_SHIFT_LAST`.  We can do a separate commit to change everything to `MPIDI_ACCU_OP_SHIFT__*`.

**Comment by pavanbalaji** [comment]
@wesbland  Why do you need the iov structures to be packed?  Generally packed structures are not architecture portable.

File: src/mpid/ch4/include/mpidpre.h (hunk 1 above)
Comment (wesbland) [reply src/mpid/ch4/include/mpidpre.h]: Done

**Comment by wesbland** [comment]
> @wesbland Why do you need the iov structures to be packed? Generally packed structures are not architecture portable.

@pavanbalaji and I had a discussion offline about whether or not this was portable. There is concern about its availability on some platforms. We agreed to move this to another PR so we can merge the other 13 patches here that are non-controversial.

**Comment by wesbland** [comment]
test:jenkins/ch4/most
test:jenkins/ch3/most

**Comment by wesbland** [comment]
I _believe_ this is passing the same number of tests as the nightlies, right?

Can we merge this @pmodels/pmodels-core?

**Comment by pavanbalaji** [comment]
This is impossible to parse.  What happened to @hzhou's collective fixes?  I thought he reverting the bad commits for now.

**Comment by hzhou** [comment]
The fix #3605 is ready to merge, only waiting for the final tests to come back. This PR could wait until that fix gets merged first. I think we are being cautious this time :)

**Comment by pavanbalaji** [comment]
Thanks.  @wesbland let's wait till @hzhou's fix gets in and then rerun the tests.

**Comment by wesbland** [comment]
Fine with me.

**Comment by pavanbalaji** [comment]
test:jenkins/ch4/most
test:jenkins/ch3/most

**Comment by pavanbalaji** [comment]
@wesbland 

What's the point of this commit: `coll: Remove malloc macro usage`?  Instead, I'd have recommended asserting in `fn_fail` instead of silently returning a bad value (although, I agree with the commit message that returning error codes would be better).

The other commit that I didn't yet signoff on (`ch4: Move symheap variable`) looks fine.  You can add my signoff on that one.

Tests seem to look good, but once #3606 is merged, we should rebase on to `mpich/master` and rerun the `ch3:mxm` tests once again.

**Comment by pavanbalaji** [comment]
test:jenkins/ch3/mxm

**Comment by pavanbalaji** [comment]
I've rerun the `mxm` tests and they look good.  The only thing remaining is my comment above for the `coll: Remove malloc macro usage` commit.

**Comment by wesbland** [comment]
Asserting at the label is fine if that’s your preference. I’m not going to be on my computer again til the morning to change it though.

**Comment by wesbland** [comment]
I made the changes. Sanity checking one more time before merging.

test:jenkins/ch3/mxm

**Comment by pavanbalaji** [comment]
The updated patch looks good to me.  I'll go ahead and approve this PR.  Did you look at the patches that I added?

**Comment by wesbland** [comment]
Yep. Looks good to me. 👍 

Do we need another signoff here or should I merge it when the tests are done? I can add signoffs from you and me to the patches at the end.

**Comment by pavanbalaji** [comment]
You can merge it.  Tests look good.
